src/main/java/org/opensearch/security/filter/NettyRequest.java

```diff
@@ -86,18 +88,12 @@ public String uri() {
 
     @Override
     public Map<String, String> params() {
-        return params(underlyingRequest.uri());
+        return parameters.get();
     }
 
     @Override
-    public String param(String key) {
-        Map<String, String> urlParams = params();
-        consumedParams.add(key);
-        return urlParams != null ? params().get(key) : null;
-    }
-
-    public Set<String> getConsumedParams() {
-        return consumedParams;
+    public Set<String> getUnconsumedParams() {
```

Comment:
Shouldn't this method remain as `getConsumedParams()`? The method returns the keys that have been accessed.

Reply:
I ran into this when thinking about the interface - I think parameters are either unconsumed and consumed; furthermore unconsumed parameters need to be consumed by something.  This makes the overall design to transition parameters from accessed -> unconsumed -> consumed.

Since OpenSearchRequest has a way to consume parameters, returning an empty set is correct, whereas if it was accessed or consumed parameters it would return a list to be actioned on - which isn't needed.

I'd prefer the new name, but don't want to hold up your on this topic. What do you think about this?

Reply:
> This makes the overall design to transition parameters from accessed -> unconsumed -> consumed

I see what you are saying. I'm getting confused since it clashes with the RestRequest object where these are referred to as consumed params, but in this case they are accessed in the header verifier, but finally consumed in the SecurityRestFilter. I wonder if we should refer to them as accessed params to avoid a naming clash, but I see what you mean now.